junifer/markers/collection.py

```diff
@@ -131,14 +131,19 @@ def validate(self, datagrabber: "BaseDataGrabber") -> None:
         logger.info(f"Data Reader output type: {t_data}")
 
         if self._preprocessors is not None:
+            validated_input_data_types = []
             for preprocessor in self._preprocessors:
                 logger.info(
                     "Validating Preprocessor: "
                     f"{preprocessor.__class__.__name__}"
                 )
                 # Validate preprocessor
-                t_data = preprocessor.validate(t_data)
-                logger.info(f"Preprocess output type: {t_data}")
+                p_data = preprocessor.validate(t_data)
+                logger.info(f"Preprocess output type: {p_data}")
+                # Extend validated list
+                validated_input_data_types.extend(p_data)
+            # Set t_data for further use
+            t_data = list(set(validated_input_data_types) | set(t_data))
```

Comment:
I still think this needs to be done inside the for. What if a preprocessor adds a new type than then is used by another preprocessor?

Reply:
Fair argument, have updated with the code you posted.